aws/s3_test.go

```diff
@@ -185,7 +186,14 @@ func testListS3Bucket(t *testing.T, args TestListS3BucketArgs) {
 	// Please note that we are passing the same session that was used to create the bucket
 	// This is required so that the defer cleanup call always gets the right bucket region
 	// to delete
-	defer nukeAllS3Buckets(awsParams.awsSession, []*string{aws.String(bucketName)}, 1000)
+	defer func() {
+		_, err := nukeAllS3Buckets(awsParams.awsSession, []*string{aws.String(bucketName)}, 1000)
+		if args.shouldError {
+			assert.Error(t, err)
+		} else {
+			assert.NoError(t, err)
```

Comment:
Does `assert` work in a separate goroutine? Or can you only call these in the original goroutine used by the test?

Reply:
I'm a bit confused: which goroutine are you referring?

Reply:
Oh, scratch that, I think I just had a brain fart and misread `defer func() { ... }` as `go func() { ... }`.